### ISSUE TITLE
Does not build on current Fedora 23

### ISSUE THREAD

**Issue by martin-langhoff** [evpo/EncryptPad#7]
Attempting to build on Fedora 23, from current `master`, with some hints from https://www.reddit.com/r/crypto/comments/488huo/encryptpad_is_an_open_source_openpgp_symmetric/

* On top of the `Development Tools` group, installed `clang` and `q5-atbase-devel`.
* Modified `build/Makefile.qt_ui` to use `qmake-qt5` (exporting QMAKE did not work)
* in cli/Makefile removed LDFLAGS += -static

However the build still errors out with (using `make --debug=b`):

```
    Must remake target 'release/EncryptPad'.
make[2]: Entering directory '/home/martin/src/EncryptPad/build/qt_build'
g++ -Wl,-O1 -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -o release/EncryptPad release/obj/application.o release/obj/file_name_helper.o release/obj/main.o release/obj/mainwindow.o release/obj/set_key_dialog.o release/obj/set_password_dialog.o release/obj/async_load.o release/obj/plain_text_switch.o release/obj/file_properties_dialog.o release/obj/new_key_dialog.o release/obj/preferences_dialog.o release/obj/recent_files_service.o release/obj/find_and_replace.o release/obj/get_password_dialog.o release/obj/confirm_password_dialog.o release/obj/get_password_or_key_dialog.o release/obj/set_encryption_key.o release/obj/file_encryption_dialog.o release/obj/m_window_load_adapter.o release/obj/file_encryption_dlg_adapter.o release/obj/file_dlg_async.o release/obj/password_generation_dialog.o release/obj/load_save_handler.o release/obj/file_request_service.o release/obj/plain_text_edit.o release/obj/find_dialog.o release/obj/qrc_EncryptPad.o release/obj/moc_application.o release/obj/moc_mainwindow.o release/obj/moc_set_key_dialog.o release/obj/moc_set_password_dialog.o release/obj/moc_async_load.o release/obj/moc_plain_text_switch.o release/obj/moc_file_properties_dialog.o release/obj/moc_new_key_dialog.o release/obj/moc_preferences_dialog.o release/obj/moc_recent_files_service.o release/obj/moc_find_and_replace.o release/obj/moc_get_password_dialog.o release/obj/moc_confirm_password_dialog.o release/obj/moc_get_password_or_key_dialog.o release/obj/moc_file_encryption_dialog.o release/obj/moc_file_dlg_async.o release/obj/moc_password_generation_dialog.o release/obj/moc_plain_text_edit.o release/obj/moc_find_dialog.o   -L/home/martin/src/EncryptPad/qt_ui/../deps/stlplus/portability/GNULINUX-x86_64-release -L/home/martin/src/EncryptPad/qt_ui/../deps/botan -L/home/martin/src/EncryptPad/qt_ui/../back_end_src/GNULINUX-x86_64-release -lback_end_src -lportability -lbotan-1.10 /home/martin/src/EncryptPad/qt_ui/../deps/zlib/libz.a -lQt5Widgets -lQt5Gui -lQt5Core -lGL -lpthread
/usr/bin/ld: /home/martin/src/EncryptPad/qt_ui/../back_end_src/GNULINUX-x86_64-release/libback_end_src.a(password_generator.o): relocation R_X86_64_32 against `.bss' can not be used when making a shared object; recompile with -fPIC
/home/martin/src/EncryptPad/qt_ui/../back_end_src/GNULINUX-x86_64-release/libback_end_src.a: error adding symbols: Bad value
collect2: error: ld returned 1 exit status
Makefile:258: recipe for target 'release/EncryptPad' failed
make[2]: *** [release/EncryptPad] Error 1
make[2]: Leaving directory '/home/martin/src/EncryptPad/build/qt_build'
Makefile.qt_ui:28: recipe for target 'x_make' failed
make[1]: *** [x_make] Error 2
make[1]: Leaving directory '/home/martin/src/EncryptPad/build'
Makefile:3: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by martin-langhoff** [comment]
I am a Fedora packager, if I can get this to build, and is usable I will probably get it into Fedora.

**Comment by evpo** [comment]
I have made the Linux build a bit friendlier since [0.3.2.1 release](https://github.com/evpo/EncryptPad/releases/tag/v0.3.2.1). You can clone from master:

    git clone https://github.com/evpo/EncryptPad.git

However, I will test it on Fedora when I have a chance to see if there are any specific issues.

**Comment by martin-langhoff** [comment]
Yes, all my tests were on `master` on the latest Fedora release, which has all the dependencies as far as I can tell.

**Comment by martin-langhoff** [comment]
It builds, it works! Thank you! Would be good to add to the README under the compiling section.

You don't need `kernel-devel`, `kernel-headers` nor `dkms`.

How hard would it be to fix the compile to use dynamic linking?

**Comment by martin-langhoff** [comment]
Thanks for your help, feel free to close this ticket, or we can continue discussion re dynamic linking and packaging for Fedora.

**Comment by evpo** [comment]
yes, `kernel-devel`, `kernel-headers` and `dkms` were not for `EncryptPad`. I installed them for `VirtualBox guest additions` and forgot to clean up the list. The same applies to `bzip2` and `perl`. Everything else I installed one by one when building. I will update the README file.

Regarding dynamic linking, it should not be hard to support it and I will make necessary changes when time allows. I configured static linking for Windows because I wanted to have one exe file without dlls for convenience. It is not needed for Linux and Mac. I can make it conditional.

I wanted to have an easy to install package myself because it's not easy sometimes for a non-technical user to build EncryptPad. Are there other requirements besides dynamic linking? Is it OK to include `zlib` and `botan` sources or preferred to use packages available on the system? I will have a look at [this document](https://fedoraproject.org/wiki/Packaging:Guidelines) as well.

**Comment by martin-langhoff** [comment]
Good questions. Yes, you're expected to link against the system botan and zlib -- does not need to be the default; as long as I can pass `--use-system-botan --use-system-zlib` (or something to the same effect) to your `configure.sh` we are in business.

The packaging guidelines mostly affect me, but yes it's fantastic if you give them a read and we can work together to ensure the rpm package can follow them. Debian's needs are pretty much the same, so the homework only needs to be done once :-)

Closing this one, as it's fixed, and you've created #10 to cover further work. Thank you!

**Comment by evpo** [comment]
@martin-langhoff 
You mentioned possibility of making a Fedora package. I would be very interested to continue with that. Is there anything else I could do to make it happen?
